Cargo.toml

```diff
@@ -11,9 +11,15 @@ edition = "2018"
 build = "build.rs"
 
 [dependencies]
+bytes = { version = "1", optional = true }
 hashbrown = "0.15"
+http = { version = "1", optional = true }
 log = "0.4"
 
+[features]
+default = []
+header-value = ["dep:bytes", "dep:http"]
```

Comment:
This feature could be renamed to `valid-header`, `strict-header` or `rfc-header` to make this reusable in case we want to add restrictions on header name in the future.

Alternatively, we could drop this as a feature and instead add those functions as variants with `_typed()` suffix.

Thoughts?